src/search.ts

```diff
@@ -229,6 +195,22 @@ export const getChosenGifUrl: (
 				case 'url':
 					// resolve the promise to the url of the picture
 					resolve(message.text);
+					// update the global state with the new Search
+					const prevHistory: HistoryEntry[] | undefined = state.get('history');
+					const newEntry: HistoryEntry = new HistoryEntry(
+						message.text.label,
+						message.text.url
+					);
+					const nextHistory =
+						// TODO: the 15 could be a setting
+						prevHistory && prevHistory.length < 15
+							? [newEntry].concat(prevHistory)
+							: prevHistory
+							? [newEntry].concat(prevHistory.slice(0, -1))
+							: [newEntry];
```

Comment:
not high priority / critical but linter usually forbif nesting ternary operators.
Maybe we could just do 
```
const nextHistory = [newEntry].concat(prevHistory || []);
if(nextHistory.length >=15){
nextHistory.pop();
}
```